test/unit/src/index-paginator-test.mjs

```diff
@@ -432,6 +439,356 @@ test('Async iterator - plugin', async t => {
+    t.equal(requests[1].body.Cursor, expectedCursor, 'Request cursor matches previous response token')
+
+    // Third page
+    expectedCursor = nestedResponseBodies[1].Nest.Token
+    t.equal(requests[2].body.Cursor, expectedCursor, 'Request cursor matches previous response token')
+
+    // Correct number of requests
+    t.equal(requests.length, simpleResponses.length, 'Server does not send extra request')
+    reset()
+  })
+
+  t.test('Headers cursor', async t => {
+    t.plan(4)
+    const responseGenerator = makeResponses(simpleResponses)
+    let expectedCursor
+    expectedResponsePayload = { Accumulator: [ 'a1', 'a2', 'a3' ] }
+
+    server.use({ accumulateRequests: true })
+    server.use({ responseGenerator })
+
+    response = await aws({
+      service,
+      headers,
+      payload: {},
+      paginate: true,
+      paginator: {
+        cursor: 'Cursor',
+        token: 'Token',
+        accumulator: 'Accumulator',
+        type: 'headers',
+      },
+    })
+
+    // Response payload
+    requests = server.getCurrentRequest()
+    t.deepEqual(response.payload, expectedResponsePayload, 'Response is correct')
+
+    // Second page
+    expectedCursor = simpleResponseBodies[0].Token
+    t.equal(requests[1].headers.cursor, expectedCursor, 'Request cursor matches previous response token')
+
+    // Third page
+    expectedCursor = simpleResponseBodies[1].Token
+    t.equal(requests[2].headers.cursor, expectedCursor, 'Request cursor matches previous response token')
+
+    // Correct number of requests
+    t.equal(requests.length, simpleResponses.length, 'Server does not send extra request')
+    reset()
+  })
+
+  t.test('Headers cursor - nested', async t => {
+    t.plan(4)
+    const responseGenerator = makeResponses(nestedResponses)
+    let expectedCursor
+    expectedResponsePayload = { Nest: { Accumulator: [ 'a1', 'a2', 'a3' ] } }
+
+    server.use({ accumulateRequests: true })
+    server.use({ responseGenerator })
+
+    response = await aws({
+      service,
+      headers,
+      paginate: true,
+      paginator: {
+        cursor: 'Cursor',
+        token: 'Nest.Token',
+        accumulator: 'Nest.Accumulator',
+        type: 'headers',
+      },
+    })
+    requests = server.getCurrentRequest()
+
+    // Response payload
+    t.deepEqual(response.payload, expectedResponsePayload, 'Response is correct')
+
+    // Second page
+    expectedCursor = nestedResponseBodies[0].Nest.Token
+    t.equal(requests[1].headers.cursor, expectedCursor, 'Request cursor matches previous response token')
+
+    // Third page
+    expectedCursor = nestedResponseBodies[1].Nest.Token
+    t.equal(requests[2].headers.cursor, expectedCursor, 'Request cursor matches previous response token')
+
+    // Correct number of requests
+    t.equal(requests.length, nestedResponseBodies.length, 'Server does not send extra request')
+    reset()
+  })
+})
+
+test('Default paginator - plugin', async t => {
```

Comment:
I think we should bail on using a production plugin for this test case, and replicate the use case with a stripped down custom plugin, as we do in other tests (e.g. https://github.com/architect/aws-lite/blob/main/test/unit/src/index-plugins-test.mjs).